test/test.js

```diff
@@ -750,6 +750,11 @@ describe('Page', function() {
       expect(requests.length).toBe(1);
       expect(requests[0].url).toBe(EMPTY_PAGE);
     }));
+    it('should navigate to URL with authentication set but no challenge', SX(async function() {
```

Comment:
I've been limited with testing because the SimpleServer class doesn't seem to support authentication.

Reply:
The testing is very important.
1. Let's add basic HTTP auth for the SimpleServer:

```js
server.setAuth(route, user, pass)
```
It should be a matter of one HTTP header.
(Make sure to reset the auth at `SimpleServer.reset` so that tests start with clean state.)

2. Let's add at least the following tests:
- Basic `page.authenticate` 
- Check what happens in case of invalid credentials
- `page.authenticate` works with cross-process navigation (look for `CROSS_PROCESS_PREFIX`)
- `page.authenticate` works with redirects
- Make sure `page.authenticate` doesn't allow calling `request.continue()`
- Make sure `page.authenticate` and `page.setRequestInterceptionEnabled(true)` work nice together